README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+
+To implement these events, just implement the following interfaces. You can implement as many as you want:
+
+* [BeforeInserter](https://godoc.org/github.com/src-d/go-kallax#BeforeInserter)
+* [BeforeUpdater](https://godoc.org/github.com/src-d/go-kallax#BeforeUpdater)
+* [BeforeSaver](https://godoc.org/github.com/src-d/go-kallax#BeforeSaver)
+* [BeforeDeleter](https://godoc.org/github.com/src-d/go-kallax#BeforeDeleter)
+* [AfterInserter](https://godoc.org/github.com/src-d/go-kallax#AfterInserter)
+* [AfterUpdater](https://godoc.org/github.com/src-d/go-kallax#AfterUpdater)
+* [AfterSaver](https://godoc.org/github.com/src-d/go-kallax#AfterSaver)
+* [AfterDeleter](https://godoc.org/github.com/src-d/go-kallax#AfterDeleter)
+
+Example:
+
+```go
+func (u *User) BeforeSave() error {
+        if u.Password == "" {
+                return errors.New("cannot save user without password")
+        }
+
+        if !isCrypted(u.Password) {
+                u.Password = crypt(u.Password)
+        }
+        return nil
+}
+```
+
+## Kallax generated code
+
+Kallax generates a bunch of code for every single model you have and saves it to a file named `kallax.go` in the same package.
+
+For every model you have, kallax will generate the following for you:
+
+* Internal methods for your model to make it work with kallax and satisfy the [Record](https://godoc.org/github.com/src-d/go-kallax#Record) interface.
+* A store named `{TypeName}Store`: the store is the way to access the data. A store of a given type is the way to access and manipulate data of that type.
+* A query named `{TypeName}Query`: the query is the way you will be able to build programmatically the queries to perform on the store. A store only will accept queries of its own type.
+The query will contain methods for adding criteria to your query for every field of your struct, called `FindBy`s.
+* A resultset named `{TypeName}ResultSet`: a resultset is the way to iterate over and obtain all elements in a resultset returned by the store. A store of a given type will always return a result set of the matching type, which will only return records of that type.
+* Schema of all the models containing all the fields. That way, you can access the name of a specific field without having to use a string, that is, a typesafe way.
+
+## Model schema
+
+### Automatic schema generation and migrations
+
+Automatic `CREATE TABLE` for models and migrations is not yet supported, even though it will probably come in future releases.
+
+### Use schema
+
+A global variable `Schema` will be created in your `kallax.go`, that contains a field with the name of every of your models. Those are the schemas of your models. Each model schema contains all the fields of that model.
+
+So, to access the username field of the user model, it can be accessed as:
+
+```go
+Schema.User.Username
+```
+
+## Manipulate models
+
+For all of the following sections, we will assume we have a store `store` for our model's type.
```

Comment:
In the following examples we provide implementations for save, update... and we just use the store, but it is not said how to obtain it.
A possible example for this section could be:
```userStore := NewUserStore()```
Or just add the this store definition in the following save/update/insert examples